### PR TITLE
[zh] Migrating telemetry and security agents from dockershim

### PR DESCRIPTION
**Pull request by X723166587**
Apply PR #31607 corrections to Chinese content.

### CODE REVIEW

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 1]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+The Pod accessing Docker may be named something like:
+
+- `signalfx-agent`
+-->
+
+SignalFx Smart Agent（已弃用）在 Kubernetes 集群上使用了多种不同的监视器，
+包括 `kubernetes-cluster`，`kubelet-stats/kubelet-metrics`，`docker-container-stats`。
+`kubelet-stats` 监视器此前已被弃用，现支持 `kubelet-metrics` 。
+`docker-container-stats` 监视器受 dockershim 移除的影响。
+`docker-container-stats` 监视器不要使用 Docker Engine 之外的运行时。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:287]:
```suggestion
不要为 `docker-container-stats` 监视器使用 Docker Engine 之外的运行时。
```

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 2]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+
+如何从依赖 dockershim 的代理迁移：
+1. 从[所配置的监视器](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitor-config.md)中移除 `docker-container-stats`。
+   注意，若节点上已经安装了 Docker，在非 dockershim 环境中启用此监视器后会导致报告错误的指标；
+   如果节点未安装 Docker，则无法获得指标。
+2. [启用和配置 `kubelet-metrics`](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/kubelet-metrics.md) 监视器。
+
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:295]: 删除此空行，留一个

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 3]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+{{< note >}}
+收集的指标会发生变化。具体请查看你的告警规则和仪表板。
+{{< /note >}}
+
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:300]: 删除此空行，留一个

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 4]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+<!--
+Flame does not support container runtimes other than Docker. See
+[https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
+-->
+
+Flame 不支持 Docker 以外的容器运行时，具体可见[https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
+
+
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:314]: 删除此空行，留一个

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 5]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+名字中包含以下字符串的 Pod 可能访问 Docker：
+- `signalfx-agent`
+
+### Yahoo Kubectl Flame
+
+<!--
+Flame does not support container runtimes other than Docker. See
+[https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:310]: 删除空行310

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 6]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+{{< note >}}
+The set of collected metrics will change. Review your alerting rules and dashboards.
+{{< /note >}}
+
+The Pod accessing Docker may be named something like:
+
+- `signalfx-agent`
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:282]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 7]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+### [Prisma Cloud Compute](https://docs.paloaltonetworks.com/prisma/prisma-cloud.html)
+
+<!--
+Check [documentation for Prisma Cloud](https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-admin-compute/install/install_kubernetes.html),
+under the "Install Prisma Cloud on a CRI (non-Docker) cluster" section.
+The pod accessing Docker may be named like:
+- `twistlock-defender-ds`
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:250]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 8]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+<!--
+How to migrate:
+
+[Migrate Falco from dockershim](https://falco.org/docs/getting-started/deployment/#docker-deprecation-in-kubernetes)
+Falco supports any CRI-compatible runtime (containerd is used in the default configuration); the documentation explains all details.
+The pod accessing Docker may have name containing: 
+- `falco`
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:233]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 9]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+Containerd support announcement: [Get automated full-stack visibility into
+containerd-based Kubernetes
+environments](https://www.dynatrace.com/news/blog/get-automated-full-stack-visibility-into-containerd-based-kubernetes-environments/)
+
+CRI-O support announcement: [Get automated full-stack visibility into your CRI-O Kubernetes containers (Beta)](https://www.dynatrace.com/news/blog/get-automated-full-stack-visibility-into-your-cri-o-kubernetes-containers-beta/)
+
+The pod accessing Docker may have name containing: 
+- `dynatrace-oneagent`
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:213]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 10]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+<!--
+How to migrate:
+[Docker deprecation in Kubernetes](https://docs.datadoghq.com/agent/guide/docker-deprecation/)
+The pod that accesses Docker Engine may have a name containing any of:
+
+- `datadog-agent`
+- `datadog`
+- `dd-agent`
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:189]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 11]
```diff
@@ -147,11 +151,164 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
 请与供应商联系，获取从 dockershim 迁移的最新说明。
+
+## 从 dockershim 迁移 {#migration-from-dockershim}
+
+### [Aqua](https://www.aquasec.com)
+
+<!--
+No changes are needed: everything should work seamlessly on the runtime switch.
+-->
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:175]: 删除此空行

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 12]
```diff
@@ -47,15 +49,16 @@ such as a pod name, is only available from Kubernetes components. Other data, su
 metrics, is not the responsibility of the container runtime. Early telemetry agents needed to query the
 container runtime **and** Kubernetes to report an accurate picture. Over time, Kubernetes gained
 the ability to support multiple runtimes, and now supports any runtime that is compatible with
-the container runtime interface.
+the [container runtime interface](/docs/concepts/architecture/cri/).
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:53]: 删除此空行

**Review by tengqm — COMMENTED** [review]
please squash your commits.

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 13]
```diff
@@ -147,11 +150,153 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+名字中包含以下字符串的 Pod 可能访问 Docker：
+- `signalfx-agent`
+
+### Yahoo Kubectl Flame
+
+<!--
+Flame does not support container runtimes other than Docker. See
+[https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
+-->
+Flame 不支持 Docker 以外的容器运行时，具体可见[https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
```
Comment (Sea-n) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:301]:
```suggestion
Flame 不支持 Docker 以外的容器运行时，具体可见 [https://github.com/yahoo/kubectl-flame/issues/51](https://github.com/yahoo/kubectl-flame/issues/51)
```

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 14]
```diff
@@ -147,11 +150,153 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+The set of collected metrics will change. Review your alerting rules and dashboards.
+{{< /note >}}
+
+The Pod accessing Docker may be named something like:
+
+- `signalfx-agent`
+-->
+SignalFx Smart Agent（已弃用）在 Kubernetes 集群上使用了多种不同的监视器，
+包括 `kubernetes-cluster`，`kubelet-stats/kubelet-metrics`，`docker-container-stats`。
+`kubelet-stats` 监视器此前已被弃用，现支持 `kubelet-metrics` 。
```
Comment (Sea-n) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:278]:
```suggestion
`kubelet-stats` 监视器此前已被弃用，现支持 `kubelet-metrics`。
```

**Review by Sea-n — COMMENTED** [review]
Looks good to me, just a few nitpick.

**Comment by tengqm** [comment]
/approve
/hold
Good to go once commits are squashed.

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 15]
```diff
@@ -147,11 +150,153 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+The set of collected metrics will change. Review your alerting rules and dashboards.
+{{< /note >}}
+
+The Pod accessing Docker may be named something like:
+
+- `signalfx-agent`
+-->
+SignalFx Smart Agent（已弃用）在 Kubernetes 集群上使用了多种不同的监视器，
+包括 `kubernetes-cluster`，`kubelet-stats/kubelet-metrics`，`docker-container-stats`。
+`kubelet-stats` 监视器此前已被弃用，现支持 `kubelet-metrics`。
```
Comment (howieyuen) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:278]:
```suggestion
`kubelet-stats` 监视器此前已被供应商所弃用，现支持 `kubelet-metrics`。
```

File: content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md [hunk 16]
```diff
@@ -147,11 +150,153 @@ telemetry agents on the node, make sure to check with the vendor of the agent wh
...
+不要为 `docker-container-stats` 监视器使用 Docker Engine 之外的运行时。
+
+如何从依赖 dockershim 的代理迁移：
+1. 从[所配置的监视器](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitor-config.md)中移除 `docker-container-stats`。
+   注意，若节点上已经安装了 Docker，在非 dockershim 环境中启用此监视器后会导致报告错误的指标；
+   如果节点未安装 Docker，则无法获得指标。
+2. [启用和配置 `kubelet-metrics`](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/kubelet-metrics.md) 监视器。
+
+{{< note >}}
+收集的指标会发生变化。具体请查看你的告警规则和仪表板。
```
Comment (howieyuen) [inline content/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md:289]:
```suggestion
收集的指标会发生变化。具体请查看你的告警规则和仪表盘。
```

**Review by howieyuen — CHANGES_REQUESTED** [review]

**Comment by Sea-n** [comment]
Hey @X723166587,

Please squash your changes into one commit.
You need to download local git client to do it.

**Review by howieyuen — APPROVED** [review]
/lgtm

**Pull request merged**